packages/core/android/src/test/kotlin/com/datadog/reactnative/DdRumTest.kt

```diff
@@ -0,0 +1,407 @@
+
+    @Mock
+    lateinit var mockRumMonitor: RumMonitor
+
+    @Mock
+    lateinit var mockReactContext: ReactApplicationContext
+
+    var mockPromise: Promise = PromiseImpl(null, null)
+
+    lateinit var fakeContext: Map<String, Any?>
+
+    @LongForgery(1000000000000L, 2000000000000L)
+    var fakeTimestamp: Long = 0L
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        GlobalRum.registerIfAbsent(mockRumMonitor)
+
+        fakeContext = forge.aMap {
+            anAlphabeticalString() to aNullable {
+                anElementFrom(
+                    anHexadecimalString(),
+                    anInt(),
+                    aLong(),
+                    getForgery<Date>(),
+                    getForgery<Throwable>()
+                )
+            }
+        }
+
+        testedDdRum = DdRum(mockReactContext)
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        GlobalRum.javaClass.setStaticValue("monitor", mock<RumMonitor>())
+        GlobalRum.javaClass.getStaticValue<GlobalRum, AtomicBoolean>("isRegistered").set(false)
+    }
+
+    @Test
+    fun `M call startView W startView()`(
+        @StringForgery key: String,
+        @StringForgery name: String
+    ) {
+        // Given
+        val updatedContext = fakeContext.toReadableMap().toHashMap().toMutableMap().apply {
+            put(RumAttributes.INTERNAL_TIMESTAMP, fakeTimestamp)
+        }
+
+        // When
+        testedDdRum.startView(key, name, fakeContext.toReadableMap(), fakeTimestamp, mockPromise)
+
+        // Then
+        verify(mockRumMonitor).startView(key, name, updatedContext)
+    }
+
+    @Test
+    fun `M call stopView W stopView()`(
+        @StringForgery key: String
+    ) {
+        // Given
+        val updatedContext = fakeContext.toReadableMap().toHashMap().toMutableMap().apply {
+            put(RumAttributes.INTERNAL_TIMESTAMP, fakeTimestamp)
+        }
+
+        // When
+        testedDdRum.stopView(key, fakeContext.toReadableMap(), fakeTimestamp, mockPromise)
+
+        // Then
+        verify(mockRumMonitor).stopView(key, updatedContext)
+    }
+
+    @Test
+    fun `M call addAction W addAction()`(
+        @StringForgery name: String,
+        @Forgery type: RumActionType
+    ) {
+        // Given
+        val updatedContext = fakeContext.toReadableMap().toHashMap().toMutableMap().apply {
+            put(RumAttributes.INTERNAL_TIMESTAMP, fakeTimestamp)
+        }
+
+        // When
+        testedDdRum.addAction(
+            type.name, name, fakeContext.toReadableMap(),
+            fakeTimestamp, mockPromise
+        )
+
+        // Then
+        verify(mockRumMonitor).addUserAction(type, name, updatedContext)
+    }
+
+    @Test
+    fun `M call addAction W addAction() with invalid type`(
+        @StringForgery name: String,
+        @StringForgery(StringForgeryType.HEXADECIMAL) type: String
+    ) {
+        // Given
+        val updatedContext = fakeContext.toReadableMap().toHashMap().toMutableMap().apply {
```

Comment:
`fakeContext` is already has type `Map`, so this can be simplified to:

```suggestion
        val updatedContext = fakeContext.toMutableMap().apply {
```

same applies to the similar lines below